### ISSUE TITLE
weird logic concerning lovers.

### ISSUE THREAD

**Issue by Eldinnie** [parabola949/Werewolf#159]
So recently I played a game with the following setup
lovers were a wolf and the blacksmith. During a lynch vote the wolf was killed.
Naturally the blacksmith died aswell. However the game declared both as winners. See the following chatlog:

> Werewolf Moderator, [08.11.16 21:05]
> The villagers have cast their votes, amid doubts and suspicions. <play a> is dead. <play a> was a Werewolf 🐺
> 
> Werewolf Moderator, [08.11.16 21:05]
> With <play a>'s dying breath, <play b> rushes over and falls, crying.  They cannot stand the loss of their loved one, and shoot themselves with the nearest pistol they can find. <play b> was the Blacksmith ⚒
> 
> Werewolf Moderator, [08.11.16 21:05]
> The villagers have won! All of the wolves have died...
> 
> Werewolf Moderator, [08.11.16 21:05]
> Players Alive: 6 / 9
> 1: 💀 Dead - the Mayor 🎖 Won
> <play a>: 💀 Dead - a Werewolf 🐺❤️ Won
> <play b>: 💀 Dead - the Blacksmith ⚒❤️ Won
> 1: 🙂 Alive - a Mason 👷 Won
> 2: 🙂 Alive - a Villager 👱 Won
> 3: 🙂 Alive - Cupid 🏹 Won
> 4: 🙂 Alive - a Villager 👱 Won
> 5: 🙂 Alive - a Villager 👱 Won
> 6: 🙂 Alive - the Beholder 👁 Won
> 
> Game Length: 00:04:24

This seems to me to be very wrong. The lovers should only win if both are alive. One of them dead should never happen (because of their roles) and you can only win if you are alive or your team is winning.

This could have two causes in my mind:
1. The order of game logic on a characters death is wrong (first check win condition the check lovers so win is declared before it;s registered the lover is dead)
2. the team for lovers isn't set as an apart team. in this case the blacksmith (as a villager) would have won even after his death. Although I see the lovers as a seperate team that can not beenfit from other teams. So either they win because BOTH are alive at the moment a team wins (be it baddies or village) or they lose because both are dead.

EDIT: The part of the description I'm talking about from the cupid description:

> TL;DR if both lovers are alive at the end, and at least one of them was on the winning team, they both win together. Rolenames: Cupid/Lovebird

And from the anouncement you've become a lover:

> You have been struck by love. <name>'s memory will never fade in your heart. Your goal is for both of you to survive. If one of you dies, the other one will die out of sorrow.

Meta-info
played in chaos mode on bot version: 

```
Werewolf Moderator, [08.11.16 22:43]
Current Version: 4.0.6156.16761
Build time: 11/8/2016 9:18:42 AM (Central)
Node Version: 4.1.6156.18255
Node Id: e3267100-d0f4-432e-936c-761181f78679
```

**Comment by TheZeus121** [comment]
@Eldinnie I believe it is intended that lovers win even if only one of their teams win, it just isn't the lovers ending.

**Comment by Eldinnie** [comment]
@TheZeus121 do you know why this is intended? It's kinda weird and breaks with the role's description and expectations

**Comment by MatteoIlGrande** [comment]
@Eldinnie Yes I see your points. Although, in werewolf, we have decided that lovers are bounded together so they only win/lose together, so it may look weird, but as long as one of  the lovers wins, they both win.

**Comment by Eldinnie** [comment]
@MatteoIlGrande hmm, Then maybe the description should be changed because especially if you say they're "bounded together" I see that as a seperate faction. So it's extra weird that if one of them wins because team village/wolf wins they both win despite being dead. I would say they always lose because they died no matter what their old team does.

**Comment by renyhp** [comment]
Lovers are NOT a separate faction.
Also, the winning conditions, in general have nothing to do with being alive (Wolves win if they are half of population, regardless if they are alive or dead; villagers win if all the wolves, cults, and the SK are dead, regardless if they are alive or dead, etc.).
Having said that, lovers are just two people who have an additional winning condition: they win if at least one of them wins.

On the other hand, you are absolutely right about wrong descriptions having been given. We will change them ASAP.
